### ISSUE TITLE
Maximum call stack size exceeded

### ISSUE THREAD

**Issue by serapath** [mcollina/levelgraph#123]
Hey, I copied the code from the `readme` and run into a problem.
The code I'm using:

```js
var levelup           = require('levelup');
var levelgraph        = require("levelgraph");
var leveljs           = require('level-js');
var leveldb           = levelup('TEST', { db: leveljs });
var db                = levelgraph(leveldb);

db.put([{
  subject: "matteo",
  predicate: "friend",
  object: "daniele"
}, {
  subject: "daniele",
  predicate: "friend",
  object: "matteo"
}, {
  subject: "daniele",
  predicate: "friend",
  object: "marco"
}, {
  subject: "lucio",
  predicate: "friend",
  object: "matteo"
}, {
  subject: "lucio",
  predicate: "friend",
  object: "marco"
}, {
  subject: "marco",
  predicate: "friend",
  object: "davide"
}], function () {

  console.log('put suceeded');
  
  var stream = db.searchStream([{
    subject: "matteo",
    predicate: "friend",
    object: db.v("x")
  }, {
    subject: db.v("x"),
    predicate: "friend",
    object: db.v("y")
  }, {
    subject: db.v("y"),
    predicate: "friend",
    object: "davide"
  }]);
  stream.on("data", function(data) {
    // this will print "{ x: 'daniele', y: 'marco' }"
    console.log(data);
  });
});

```

How is it supposed to work? Am I doing something wrong?

**Comment by mcollina** [comment]
Which version of node are you using?
Il giorno sab 23 mag 2015 alle 00:44 Alexander Praetorius <
notifications@github.com> ha scritto:

**Comment by serapath** [comment]
`node -v` => **v1.8.1**
`npm -v` => **2.8.3**

**Comment by mcollina** [comment]
I recommend you to test it with iojs 2.0.0, as the tests are currently passing there.

**Comment by serapath** [comment]
I tried with `v2.1.0` because thats the newest version it gives me and it throws during installation.
Then I tried with `v2.0.0` and installation runs through.
Otherwise I am on `Linux Mint 17 with XFCE`

This time i get the `console.log('put suceeded');`.
But the execution of `db.searchStream` throws again:
`Uncaught RangeError: Maximum call stack size exceeded`

The code that runs over and over again is:
```js
// ...

// Mark that a method should not be used.
// Returns a modified function which warns once by default.
// If --no-deprecation is set, then it is a no-op.
exports.deprecate = function(fn, msg) {
  // Allow for deprecating things in the process of starting up.
  if (isUndefined(global.process)) {
    return function() {
      return exports.deprecate(fn, msg).apply(this, arguments);
    };
  }

// ...
```

It goes through:
* `var stream = db.searchStream([{...`
* `planner(query, function(err, newquery) {...`
*  then
*  ```js
function queryplanner(db, options) {
  return function planner(query, callback) {
    // dupes!
    var result = [].concat(query);

    async.each(query, function(q, cb) {
```
* then
* ```    async.each = function (arr, iterator, callback) {
        callback = callback || function () {};
        if (!arr.length) {
            return callback();
        }
        var completed = 0;
        _each(arr, function (x) {
```
* then
* ```js
      db.approximateSize(range.start, range.end, function(err, size) {
        if (err) {
          size = Object.keys(variablesMask(q)).length;
        }
        q.size = size;
        async.nextTick(cb);
      });
```

The `db.approximateSize` is where it goes into an infinite loop

**Comment by mcollina** [comment]
Which version of LevelUp are you using?

**Comment by serapath** [comment]
`^1.1.0`

**Comment by serapath** [comment]
I'm using it with `level-js`

**Comment by mcollina** [comment]
This is probably solved in 1.0.0. Can you please verify?
